options/options.cpp

```diff
@@ -683,15 +683,15 @@ ProverResult PonoOptions::parse_and_set_options(int argc,
           // not possible, because handled further above and exits the program
         case ENGINE: engine_ = to_engine(opt.arg); break;
         case BOUND:
-          bound_ = atoi(opt.arg);
-          if (bound_ == INT_MAX) {
+          bound_ = strtoul(opt.arg, NULL, 10);
+          if (bound_ == UINT_MAX) {
             throw PonoException("--bound must be less than "
-                                + std::to_string(INT_MAX) + ".");
+                                + std::to_string(UINT_MAX) + ".");
```

Comment:
The bound still has to be less than `INT_MAX`.

The reason is that in `pono.cpp`, there's a line:

```c
    r = prover->check_until(pono_options.bound_ + has_monitor);

```

where `has_monitor` is 1 or 0,
and `prover->check_until` takes as input an `int` type.
Therefore the bound must to less than `INT_MAX` to avoid overflow.

See also #378.